---
title: "PLT Regulome embryo"
author: "Merijn Kerstens"
citation: "Kerstens, Galinha, Hofhuis et al., 2023: Redundant PLETHORA activity promotes development of early embryonic cell lineages in Arabidopsis"
latest update: "23/03/2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

## Load required packages
```{r}
library(magrittr)
library(ggplot2)
```

## Functions
```{r}
# Convert counts to log10 TPM values
logTPM <- function(x) {return(log10(x+1))}

# Calculate the mid-point between of a range
get_midpoint <- function(range) {
  mean(as.numeric(unlist(strsplit(gsub("\\(|\\)|\\[|\\]", "", as.character(range)), ","))))
}

# Inverse intersection function
# From: https://www.r-bloggers.com/2011/11/outersect-the-opposite-of-rs-intersect-function/
outersect <- function(x, y) {
  sort(c(setdiff(x, y),
         setdiff(y, x)))
}

# Function to set pretty default plot
prettify <- theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  theme(axis.line = element_line(colour="black")) +
  theme(axis.text.y = element_text(colour="black", size = 14)) +
  theme(axis.text.x = element_text(colour = "black", size = 14)) +
  theme(axis.ticks = element_line(colour="black")) +
  theme(legend.text.align = 0) +
  theme(legend.text = element_text(size = 14)) +
  theme(axis.title.y = element_text(size = 16)) + 
  theme(axis.title.x = element_text(size = 16)) + 
  theme(legend.position = "right") +
  theme(strip.text.y = element_blank()) +
  theme(strip.background = element_blank()) +
  theme(panel.border = element_rect(colour = "black", fill=NA, size = 1))
```

## Specify paths
```{r}
# Folders
gen_path <- "/Users/Merijn/Desktop/PLT embryos/"
peak_path <- paste(gen_path,"PLT_target_expression/Regulome/consensus_peaks_with_motif/",sep = "")
RNA_path <- paste(gen_path,"PLT_target_expression/",sep = "")
scRNA_path <- paste(gen_path,"PLT_target_expression/Phloem scRNA-seq/",sep = "")
output_path <- paste(gen_path,"MS/2023 figs/R_plots/",sep = "")
```

## Load input files (specify path!)
```{r}
# PLT ChIP/DAP peaks (this study)
peak_dist <- read.table(paste(peak_path,"all_peaks_homer.txt",sep = ""), sep = '\t', header = T) %>%
  select(1,Distance.to.TSS, Nearest.Unigene)
colnames(peak_dist) <- c("peakname","distance","V1")

# PLT overexpression data (Santuari et al., 2016); supplemental data 1; tab "Compendium"; columns "AGI", and "PLT1" to "PLT7"
PLT_list <- read.csv(paste(RNA_path,"Santuari_set.csv",sep = ""), header=T)

# Root meristem RNA-seq (Li et al., 2016); NIHMS824562-supplement-9
Roots <- read.csv(paste(RNA_path,"celltypes_Lietal2016.csv",sep = ""), header = T)

# Shoot stem cell RNA-seq (Gutzat et al., 2020); dataset EV1; selected columns = "atg" (renamed to GeneID), "TPM_41768_D7+" (renamed to SAM_7D_1), "TPM_41766_D7+" (renamed to SAM_7D_2), "TPM_41772_D14+" (renamed to SAM_14D_1),	"TPM_41770_D14+" renamed to (SAM_14D_2), "TPM_41776_D35+" (renamed to SAM_35D_1),	"TPM_41774_D35+" (renamed to SAM_35D_2)
SAM <- read.csv(paste(RNA_path,"SAM_Gutzat_R.csv", sep = ""), header=T)

# Apical cell and basal cell RNA-seq (Zhou et al., 2020); AC1/2/3 and BC1/2/3 downloaded from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE135422 then merged to a 4-column GeneID-C1-C2-C3 file
AC <- read.csv(paste(RNA_path,"AC.csv", sep = ""), header=T)
BC <- read.csv(paste(RNA_path,"BC.csv", sep = ""), header=T)

# Protophloem sieve element scRNA-seq data (Roszak et al., 2021) cuffnorm FPKM; from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE140778
phloemsc <- read.table(paste(scRNA_path,"GSE140778_20190115_data.txt", sep = ""), header = T)

# Protophloem sieve element scRNA-seq data (Roszak et al., 2021); pseudotime (Table S17); selected columns = "ID", "Cluster_PSE", and "Pseudotime_PSE"
phloem_pt <- read.table(paste(scRNA_path,"phloem_scseq_pseudotime_Roszak2021.txt", sep = ""), header = T)
```

## Determine number and distribution of peaks
```{r}
### Counts ####

# Count number of annotated, overlapping peaks
length(unique(peak_dist$peakname))

# Count number of annotated, overlapping peaks in terms of unique AGIs
length(unique(peak_dist$V1))

# Count number of annotated, overlapping [-4000,4000] peaks in terms of unique AGIs
tmp <- peak_dist %>% filter(distance >= -4000 & distance <= 4000)
length(unique(tmp$V1))

### Make dataframe containing the 2406 selected peaks ###
peak_list <- peak_dist %>% 
  filter(distance >= -4000 & distance <= 4000) %>% 
  select(V1) %>% 
  unique()

### Plotting ###
ggplot() +
  geom_vline(xintercept = 0, size = 0.7, linetype = "dashed", color = "grey") +
  geom_density(data = peak_dist, aes(y = ..density.. * 10^4, x = distance), color = "black") +
  xlim(-4001,4001) +
  ylim(0, 5) +
  xlab("Distance to TSS (bp)") +
  ylab(expression(Density~(10^{"-4"}))) +
  prettify

ggsave(paste(output_path,"peaks_tss.pdf",sep = ""), width = 4, height = 4)
```

## Select upregulated genes only (not genes in upregulated by one+ PLT and downregulated by another)
```{r}
## Filter out non/downregulated genes and NAs from the microarray data
PLT_list %<>%
  mutate(regu = rowSums(.[2:7])) %>%
  filter(regu > 0) %>%
  filter(PLT1 > -1 & PLT2 > -1 & PLT3 > -1 & PLT4 > -1 & PLT5 > -1 & PLT7 > -1) %>%
  na.omit %>%
  .[1:7]

dim(PLT_list)
```

## Overlap
```{r}
## Activate this code for the complete regulome, upregulated
overlap <- intersect(PLT_list$AGI,peak_list$V1) %>% as.data.frame()
write.table(overlap, file=paste(output_path,"PLT_bound_upregulated.txt",sep = ""))

## Make new dataframe
PLT_total <- overlap
colnames(PLT_total) <- "AGI"
rownames(PLT_total) <- PLT_total$AGI
```

## Process root meristem data (Li et al., 2016) to log10 TPM
```{r}
# Sum isoforms
Roots$X %<>% gsub("\\..*$","", .)

# Extract root zones
Roots %<>%
  group_by(X) %>%
  summarise(Mer = sum(Mer), Elong = sum(Elong), Mat = sum(Mat), WOX5 = sum(WOX5))

Mer <- Roots %>% select(X, Mer)
Elong <- Roots %>% select(X, Elong)
Mat <- Roots %>% select(X, Mat)
WOX5 <- Roots %>% select(X, WOX5)

# Clean up data: FPKM should be greater than 0
dim(Mer)
Mer %<>% filter(Mer > 0) %>% column_to_rownames('X')
dim(Mer)

dim(Mat)
Mat %<>% filter(Mat > 0) %>% column_to_rownames('X')
dim(Mat)

dim(Elong)
Elong %<>% filter(Elong > 0) %>% column_to_rownames('X')
dim(Elong)

dim(WOX5)
WOX5 %<>% filter(WOX5 > 0) %>% column_to_rownames('X')
dim(WOX5)

# FPKM to log10 TPM conversion
Mer$TPM <- Mer$Mer / sum(Mer$Mer) * 1000000
Elong$TPM <- Elong$Elong / sum(Elong$Elong) * 1000000
Mat$TPM <- Mat$Mat / sum(Mat$Mat) * 1000000
WOX5$TPM <- WOX5$WOX5 / sum(WOX5$WOX5) *1000000

Mer %<>% mutate_if(is.numeric, logTPM) %>% .[2]
Elong %<>% mutate_if(is.numeric, logTPM) %>% .[2]
Mat %<>% mutate_if(is.numeric, logTPM) %>% .[2]
WOX5 %<>% mutate_if(is.numeric, logTPM) %>% .[2]
```

## Process shoot stem cell data (Gutzat et al., 2020) to log10 TPM
```{r}
# Split datasets per SAM age
rownames(SAM) <- SAM$GeneID
SAM_7D <- SAM[2:3]
SAM_14D <- SAM[4:5]
SAM_35D <- SAM[6:7]

# log10 TPM conversion 
SAM_7D %<>% mutate(avg = rowMeans(.)) %>% filter(avg > 0) %>% mutate_if(is.numeric, logTPM) %>% .[3]
SAM_14D %<>% mutate(avg = rowMeans(.)) %>% filter(avg > 0) %>% mutate_if(is.numeric, logTPM) %>% .[3]
SAM_35D %<>% mutate(avg = rowMeans(.)) %>% filter(avg > 0) %>% mutate_if(is.numeric, logTPM) %>% .[3]
```

## Process AC/BC data (Zhou et al., 2020) to log10 TPM
```{r}
# Syntax
rownames(AC) <- AC$GeneID
rownames(BC) <- BC$GeneID
AC <- AC[2:4]
BC <- BC[2:4]
dim(AC)
dim(BC)

# TPM conversion + average the three replicates, TPM, AC
AC$TPM1 <- AC$AC1 / sum(AC$AC1) * 1000000
AC$TPM2 <- AC$AC2 / sum(AC$AC2) * 1000000
AC$TPM3 <- AC$AC3 / sum(AC$AC3) * 1000000
AC <- AC[4:6]
AC %<>% mutate(avg = rowMeans(.)) %>% filter(avg > 0)

# TPM conversion + average the three replicates, TPM, BC
BC$TPM1 <- BC$BC1 / sum(BC$BC1) * 1000000
BC$TPM2 <- BC$BC2 / sum(BC$BC2) * 1000000
BC$TPM3 <- BC$BC3 / sum(BC$BC3) * 1000000
BC <- BC[4:6]
BC %<>% mutate(avg = rowMeans(.)) %>% filter(avg > 0)

# logTPM conversion
AC %<>% mutate_if(is.numeric, logTPM) %>% .[4]
dim(AC)
BC %<>% mutate_if(is.numeric, logTPM) %>% .[4]
dim(BC)
```

## Couple PLT-bound-upregulated targets to AC, BC, QC, Mer, Elong, Mat, SAM log10 TPM-transformed datasets
```{r}
# Here we extract and merge the expression values of each PLT target gene to each tissue
ACPLT_total <- PLT_total %>%
  rownames_to_column("row_names") %>%
  semi_join(rownames_to_column(AC, "row_names"), by = "row_names") %>% .[1]
tmpexpression <- AC %>% rownames_to_column("row_names")
ACexp <- merge(ACPLT_total, tmpexpression, by = "row_names")
colnames(ACexp) <- c("AGI","AC")

BCPLT_total <- PLT_total %>%
  rownames_to_column("row_names") %>%
  semi_join(rownames_to_column(BC, "row_names"), by = "row_names") %>% .[1]
tmpexpression <- BC %>% rownames_to_column("row_names")
BCexp <- merge(BCPLT_total, tmpexpression, by = "row_names")
colnames(BCexp) <- c("AGI","BC")

MerPLT_total <- PLT_total %>%
  rownames_to_column("row_names") %>%
  semi_join(rownames_to_column(Mer, "row_names"), by = "row_names") %>% .[1]
tmpexpression <- Mer %>% rownames_to_column("row_names")
Merexp <- merge(MerPLT_total, tmpexpression, by = "row_names")
colnames(Merexp) <- c("AGI","Mer")

ElongPLT_total <- PLT_total %>%
  rownames_to_column("row_names") %>%
  semi_join(rownames_to_column(Elong, "row_names"), by = "row_names") %>% .[1]
tmpexpression <- Elong %>% rownames_to_column("row_names")
Elongexp <- merge(ElongPLT_total, tmpexpression, by = "row_names")
colnames(Elongexp) <- c("AGI","Elong")

MatPLT_total <- PLT_total %>%
  rownames_to_column("row_names") %>%
  semi_join(rownames_to_column(Mat, "row_names"), by = "row_names") %>% .[1]
tmpexpression <- Mat %>% rownames_to_column("row_names")
Matexp <- merge(MatPLT_total, tmpexpression, by = "row_names")
colnames(Matexp) <- c("AGI","Mat")

WOX5PLT_total <- PLT_total %>%
  rownames_to_column("row_names") %>%
  semi_join(rownames_to_column(WOX5, "row_names"), by = "row_names") %>% .[1]
tmpexpression <- WOX5 %>% rownames_to_column("row_names")
WOX5exp <- merge(WOX5PLT_total, tmpexpression, by = "row_names")
colnames(WOX5exp) <- c("AGI","QC")

SAM7DPLT_total <- PLT_total %>%
  rownames_to_column("row_names") %>%
  semi_join(rownames_to_column(SAM_7D, "row_names"), by = "row_names") %>% .[1]
tmpexpression <- SAM_7D %>% rownames_to_column("row_names")
SAM7Dexp <- merge(SAM7DPLT_total, tmpexpression, by = "row_names")
colnames(SAM7Dexp) <- c("AGI","SAM")

SAM14DPLT_total <- PLT_total %>%
  rownames_to_column("row_names") %>%
  semi_join(rownames_to_column(SAM_14D, "row_names"), by = "row_names") %>% .[1]
tmpexpression <- SAM_14D %>% rownames_to_column("row_names")
SAM14Dexp <- merge(SAM14DPLT_total, tmpexpression, by = "row_names")
colnames(SAM14Dexp) <- c("AGI","SAM")

SAM35DPLT_total <- PLT_total %>%
  rownames_to_column("row_names") %>%
  semi_join(rownames_to_column(SAM_35D, "row_names"), by = "row_names") %>% .[1]
tmpexpression <- SAM_35D %>% rownames_to_column("row_names")
SAM35Dexp <- merge(SAM35DPLT_total, tmpexpression, by = "row_names")
colnames(SAM35Dexp) <- c("AGI","SAM")
```

## Couple 'all genes' to AC, BC, QC, Mer, Elong, Mat, SAM log10 TPM-transformed datasets
```{r}
# Here we extract and merge the expression values of all genes to each tissue; necessary for later
Mer_all <- Mer %>% rownames_to_column("AGI")
colnames(Mer_all) <- c("AGI","Mer")
Elong_all <- Elong %>% rownames_to_column("AGI")
colnames(Elong_all) <- c("AGI","Elong")
Mat_all <- Mat %>% rownames_to_column("AGI")
colnames(Mat_all) <- c("AGI","Mat")
AC_all <- AC %>% rownames_to_column("AGI")
colnames(AC_all) <- c("AGI","AC")
BC_all <- BC %>% rownames_to_column("AGI")
colnames(BC_all) <- c("AGI","BC")
WOX5_all <- WOX5 %>% rownames_to_column("AGI")
colnames(WOX5_all) <- c("AGI","QC")
SAM7D_all <- SAM_7D %>% rownames_to_column("AGI")
colnames(SAM7D_all) <- c("AGI","SAM")
SAM14D_all <- SAM_14D %>% rownames_to_column("AGI")
colnames(SAM14D_all) <- c("AGI","SAM")
SAM35D_all <- SAM_35D %>% rownames_to_column("AGI")
colnames(SAM35D_all) <- c("AGI","SAM")
```

## Expression thresholds
```{r}
## Define thresholds for expression cut off based on log10 TPM density plots; cut off was selected at 0.75, just prior to the second 'peak' in density.

logTPMthresh <- 0.75

# Upregulated
ACvenn <- ACexp %>% filter(AC > logTPMthresh)
BCvenn <- BCexp %>% filter(BC > logTPMthresh)
QCvenn <- WOX5exp %>% filter(QC > logTPMthresh)
Mervenn <- Merexp %>% filter(Mer > logTPMthresh)
Elongvenn <- Elongexp %>% filter(Elong > logTPMthresh)
Matvenn <- Matexp %>% filter(Mat > logTPMthresh)
SAM7Dvenn <- SAM7Dexp %>% filter(SAM > logTPMthresh)
SAM14Dvenn <- SAM14Dexp %>% filter(SAM > logTPMthresh)
SAM35Dvenn <- SAM35Dexp %>% filter(SAM > logTPMthresh)
SAMtotvenn <- rbind(SAM7Dvenn, SAM14Dvenn, SAM35Dvenn) %>% select(AGI) %>% unique()
```

## Expression over protophloem development, i.e. developmental pseudotime
```{r, fig.width = 1, fig.height = 2}
# Make wide format; this code takes a while
copy <- phloemsc
copy$AGI <- rownames(phloemsc)
phloemsc_long <- gather(copy, Cell_ID, norm.counts, SLX_12100_i701_i502:SLX_17313_i729_i522, factor_key=TRUE)

# Merge with pseudotime dataframe; this code takes a while so is deactivated by default
phloem <- merge(phloemsc_long, phloem_pt, by = c("Cell_ID"))
phloem_filtered <- phloem %>% filter(Pseudotime_PSE != 'NA')

# Non-PLT-regulated genes (no peak, no upregulation); 29035
agi_list <- phloem_filtered %>% select(AGI) %>% unique()
colnames(overlap) <- "AGI"
colnames(peak_list) <- "AGI"
PLT_interactions <- rbind(PLT_list[1], peak_list) %>% unique()
colnames(PLT_interactions) <- "AGI"
anti_plt_list <- anti_join(agi_list, PLT_interactions, by = "AGI")
anti_plt_sel <- merge(anti_plt_list, phloem_filtered, by = "AGI")
input_anti_plt <- anti_plt_sel %>% filter(norm.counts > 5) %>% group_by(Cell_ID, Pseudotime_PSE) %>% count()
input_anti_plt$prop <- input_anti_plt$n / 29035 * 100

# Peaks only; 2406
colnames(peak_list) <- "AGI"
PLT_interactions <- peak_list %>% unique()
colnames(PLT_interactions) <- "AGI"
peak_plt_sel <- merge(PLT_interactions, phloem_filtered, by = "AGI")
input_peak_plt <- peak_plt_sel %>% filter(norm.counts > 5) %>% group_by(Cell_ID, Pseudotime_PSE) %>% count()
input_peak_plt$prop <- input_peak_plt$n / 2406 * 100

# Upregulated genes only; 2046
PLT_interactions <- PLT_list[1] %>% unique()
colnames(PLT_interactions) <- "AGI"
upregu_plt_sel <- merge(PLT_interactions, phloem_filtered, by = "AGI")
input_upregu_plt <- upregu_plt_sel %>% filter(norm.counts > 5) %>% group_by(Cell_ID, Pseudotime_PSE) %>% count()
input_upregu_plt$prop <- input_upregu_plt$n / 2046 * 100

# Direct PLT-upregulated target genes [-4kb,+4kb]; 334
phloem_sel <- merge(PLT_total, phloem_filtered, by = "AGI")
input <- phloem_sel %>% filter(norm.counts > 5) %>% group_by(Cell_ID, Pseudotime_PSE) %>% count()
input$prop <- input$n / 334 * 100

# Perform binning with specific number of bins for each dataset, remove ranges with only one observation
input_binned <- input %>%
  group_by(Pseudotime_PSE = cut(Pseudotime_PSE, seq(-0.05, 87.05, 0.1))) %>%
  summarise(meanprop = mean(prop), n = n()) %>%
  filter(n > 1)
input_binned$midpoint <- sapply(input_binned$Pseudotime_PSE, get_midpoint)

input_peak_plt_binned <- input_peak_plt %>%
  group_by(Pseudotime_PSE = cut(Pseudotime_PSE, seq(-0.05, 87.05, 0.1))) %>%
  summarise(meanprop = mean(prop), n = n()) %>%
  filter(n > 1)
input_peak_plt_binned$midpoint <- sapply(input_peak_plt_binned$Pseudotime_PSE, get_midpoint)

input_upregu_plt_binned <- input_upregu_plt %>%
  group_by(Pseudotime_PSE = cut(Pseudotime_PSE, seq(-0.05, 87.05, 0.1))) %>%
  summarise(meanprop = mean(prop), n = n()) %>%
  filter(n > 1)
input_upregu_plt_binned$midpoint <- sapply(input_upregu_plt_binned$Pseudotime_PSE, get_midpoint)

input_anti_plt_binned <- input_anti_plt %>%
  group_by(Pseudotime_PSE = cut(Pseudotime_PSE, seq(-0.05, 87.05, 0.1))) %>%
  summarise(meanprop = mean(prop), n = n()) %>%
  filter(n > 1)
input_anti_plt_binned$midpoint <- sapply(input_anti_plt_binned$Pseudotime_PSE, get_midpoint)

## Plot all four datasets
ggplot() +
      geom_point(data = input_anti_plt_binned, aes(y = midpoint, x = meanprop), color = "#440154", size = 0.5) +
      geom_point(data = input_peak_plt_binned, aes(y = midpoint, x = meanprop), color = "#1F968B", size = 0.5) +
      geom_point(data = input_upregu_plt_binned, aes(y = midpoint, x = meanprop), color = "#39568C", size = 0.5) +
      geom_point(data = input_binned, aes(y = midpoint, x = meanprop), color = "#73D055", size = 0.5) +
      geom_smooth(data = input_anti_plt_binned, aes(y = midpoint, x = as.numeric(meanprop)), color = "#440154", method = "loess", size = 0.5, level = 0) +
      geom_smooth(data = input_peak_plt_binned, aes(y = midpoint, x = as.numeric(meanprop)), color = "#1F968B", method = "loess", size = 0.5, level = 0) +
      geom_smooth(data = input_upregu_plt_binned, aes(y = midpoint, x = as.numeric(meanprop)), color = "#39568C", method = "loess", size = 0.5, level = 0) +
      geom_smooth(data = input_binned, aes(y = midpoint, x = as.numeric(meanprop)), color = "#73D055", method = "loess", size = 0.5, level = 0) +
      xlim(0,75) +
      ylim(0,90) +
      ylab("Developmental pseudotime") +
      xlab("Percentage of expressed genes") +
      prettify
ggsave(paste(output_path,"sc_targets_4.pdf",sep=""), width = 2.5, height = 5)

# Plot only PLT-bound + upregulated vs non-PLT-bound + non-upregulated
ggplot() +
      geom_point(data = input_anti_plt_binned, aes(y = midpoint, x = meanprop), color = "#440154", size = 0.5) +
      geom_point(data = input_binned, aes(y = midpoint, x = meanprop), color = "#73D055", size = 0.5) +
      geom_smooth(data = input_anti_plt_binned, aes(y = midpoint, x = as.numeric(meanprop)), color = "#440154", method = "loess", size = 0.5, level = 0) +
      geom_smooth(data = input_binned, aes(y = midpoint, x = as.numeric(meanprop)), color = "#73D055", method = "loess", size = 0.5, level = 0) +
      xlim(0,60) +
      ylim(0,90) +
      ylab("Developmental pseudotime") +
      xlab("Percentage of expressed genes") +
      prettify
ggsave(paste(output_path,"sc_targets_2.pdf",sep=""), width = 2.5, height = 5)
```

## PLT cumulative dosage
```{r, fig.width = 1, fig.height = 2}
PLT_df <- data.frame(c("AT3G20840","AT1G51190","AT5G10510","AT5G17430","AT5G57390","AT5G65510" ))
colnames(PLT_df) <- "AGI"

cumul_PLT_input <- merge(PLT_df, phloem_filtered, by = "AGI")
cumul_PLT_input %<>% group_by(Cell_ID, Pseudotime_PSE) %>% mutate(sum = sum(norm.counts)) %>% ungroup %>% select(Pseudotime_PSE, sum) %>% unique()

# perform binning with specific number of bins
cumul_PLT_input_binned <- cumul_PLT_input %>%
  group_by(Pseudotime_PSE = cut(Pseudotime_PSE, seq(-0.05, 87.05, 0.1))) %>%
  summarise(meansum = mean(sum), n = n()) %>%
  filter(n > 1)
cumul_PLT_input_binned$midpoint <- sapply(cumul_PLT_input_binned$Pseudotime_PSE, get_midpoint)

# Plot
ggplot(data = cumul_PLT_input_binned, aes(y = midpoint, x = meansum)) +
  geom_point(color = "black", size = 0.5) +
  geom_smooth(method = "loess",  level  = 0, color = "black", size = 0.5) +
  ylab("Developmental pseudotime") +
  xlab("Cumulative PLT transcript levels") +
  ylim(0,90) +
  prettify
ggsave(paste(output_path,"plt_dosage.pdf",sep=""), width = 2.5, height = 5)
```

## Statistics (comparing phl_stem/diff between datasets, top/bottom 5 cells in terms of pseudotime)
```{r, fig.height = 4, fig.width = 3}
# Here we select the five most 'stem' or 'differentiated cells from the Roszak data.

## Stem cells
phl_stem_plt <- input %>%
  ungroup() %>%
  arrange(Pseudotime_PSE) %>%
  slice(1:5) %>%
  mutate(tissue = "stem") %>%
  mutate(dataset = "PLT") %>%
  select(Cell_ID, tissue, prop, dataset)
shapiro.test(phl_stem_plt$prop) # p-value = 0.4967

phl_stem_anti <- input_anti_plt %>%
  ungroup() %>%
  arrange(Pseudotime_PSE) %>%
  slice(1:5) %>%
  mutate(tissue = "stem") %>% 
  mutate(dataset = "anti") %>%
  select(Cell_ID, tissue, prop, dataset)
shapiro.test(phl_stem_anti$prop) # p-value = 0.1876

phl_stem_peak <- input_peak_plt %>%
  ungroup() %>%
  arrange(Pseudotime_PSE) %>%
  slice(1:5) %>%
  mutate(tissue = "stem") %>% 
  mutate(dataset = "peak") %>%
  select(Cell_ID, tissue, prop, dataset)
shapiro.test(phl_stem_peak$prop) # p-value = 0.675

phl_stem_upregu <- input_upregu_plt %>%
  ungroup() %>%
  arrange(Pseudotime_PSE) %>%
  slice(1:5) %>%
  mutate(tissue = "stem") %>% 
  mutate(dataset = "upregu") %>%
  select(Cell_ID, tissue, prop, dataset)
shapiro.test(phl_stem_upregu$prop) # p-value = 0.4524

# Thus, all 'stem' data is normally distributed

## Differentiated
phl_diff_plt <- input %>%
  ungroup() %>%
  arrange(desc(Pseudotime_PSE)) %>%
  slice(1:5) %>%
  mutate(tissue = "diff") %>% 
  mutate(dataset = "PLT") %>%
  select(Cell_ID, tissue, prop, dataset)
shapiro.test(phl_diff_plt$prop) # p-value = 0.8502

phl_diff_anti <- input_anti_plt %>%
  ungroup() %>%
  arrange(desc(Pseudotime_PSE)) %>%
  slice(1:5) %>%
  mutate(tissue = "diff") %>% 
  mutate(dataset = "anti") %>%
  select(Cell_ID, tissue, prop, dataset)
shapiro.test(phl_diff_anti$prop) # p-value = 0.3224

phl_diff_peak <- input_peak_plt %>%
  ungroup() %>%
  arrange(desc(Pseudotime_PSE)) %>%
  slice(1:5) %>%
  mutate(tissue = "diff") %>% 
  mutate(dataset = "peak") %>%
  select(Cell_ID, tissue, prop, dataset)
shapiro.test(phl_diff_peak$prop) # p-value = 0.5345

phl_diff_upregu <- input_upregu_plt %>%
  ungroup() %>%
  arrange(desc(Pseudotime_PSE)) %>%
  slice(1:5) %>%
  mutate(tissue = "diff") %>% 
  mutate(dataset = "upregu") %>%
  select(Cell_ID, tissue, prop, dataset)
shapiro.test(phl_diff_upregu$prop) # p-value = 0.8151

# Thus, all 'diff' data is normally distributed

## Merge datasets
stat_comp_phl_stem <- rbind(phl_stem_plt, phl_stem_anti, phl_stem_peak, phl_stem_upregu)
stat_comp_phl_diff <- rbind(phl_diff_plt, phl_diff_anti, phl_diff_peak, phl_diff_upregu)

## Test for equal variances
bartlett.test(prop~dataset, data = stat_comp_phl_stem) # p-value = 0.6658
bartlett.test(prop~dataset, data = stat_comp_phl_diff) # p-value = 0.2983

# Thus, both 'stem' and 'diff' sets have equal variance.

## Repeated measures anova
aov_stem <- aov(prop~dataset+Error(Cell_ID), data = stat_comp_phl_stem) %>% summary()
aov_diff <- aov(prop~dataset+Error(Cell_ID), data = stat_comp_phl_diff) %>% summary()

## Paired t-tests (multiple t-tests with Bonferroni correction)
t_stem <- pairwise_t_test(data = stat_comp_phl_stem, prop ~ dataset, paired = TRUE, p.adjust.method = "bonferroni")
t_diff <- pairwise_t_test(data = stat_comp_phl_diff, prop ~ dataset, paired = TRUE, p.adjust.method = "bonferroni")

stat_t_stem <- data.frame(
                dataset = c("PLT","peak","upregu","anti"),
                signif = c("a","b","c","d"),
                yaxis = c(60.5, 38, 69.5, 30.75)
)

stat_t_diff <-  data.frame(
                dataset = c("PLT","peak","upregu","anti"),
                signif = c("a","a","a","a"),
                yaxis = c(16, 13.75, 12.5, 12)
)

stat_comp_phl_stem$dataset <- factor(stat_comp_phl_stem$dataset, levels=c("PLT", "peak", "upregu","anti"))
stat_comp_phl_diff$dataset <- factor(stat_comp_phl_diff$dataset, levels=c("PLT", "peak", "upregu","anti"))
stat_t_stem$dataset <- factor(stat_t_stem$dataset, levels=c("PLT", "peak", "upregu","anti"))
stat_t_diff$dataset <- factor(stat_t_diff$dataset, levels=c("PLT", "peak", "upregu","anti"))

PLT <- expression("PLT-bound + PLT-upregulated")
peaks <- expression("PLT-bound")
upregu <- expression("PLT-upregulated")
anti <- expression("non-PLT-bound + non-PLT-upregulated")

# Plot all four 'stem' together
ggplot(stat_comp_phl_stem) +
  geom_violin(aes(x = dataset, y = prop, fill = dataset), color = "black", scale="width") +
  geom_jitter(aes(x = dataset, y = prop, color = dataset), position=position_jitterdodge(jitter.width = 0.2, jitter.height = 0), size = 1) +
  geom_text(data = stat_t_stem, aes(x = dataset, y = yaxis, label = signif)) +
  xlab("") +
  ylab("Percentage of expressed genes") +
  ylim(0,80) +
  prettify +
  scale_x_discrete(labels = c(PLT,peaks,upregu,anti)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_fill_manual(name="", labels = c(PLT,peaks,upregu,anti), values=c("#73D055","#1F968B","#39568C","#440154")) +
  scale_color_manual(name="", labels = c("","","","",""),values=c("black","black","black","black"))
ggsave(paste(output_path,"violin_stem_4.pdf",sep=""), width = 7, height = 7.5)

# Plot all four 'diff' together
ggplot(stat_comp_phl_diff) +
  geom_violin(aes(x = dataset, y = prop, fill = dataset), color = "black", scale="width") +
  geom_jitter(aes(x = dataset, y = prop, color = dataset),   position=position_jitterdodge(jitter.width = 0.2, jitter.height = 0), size = 1) +
  geom_text(data = stat_t_diff, aes(x = dataset, y = yaxis, label = signif)) +
  xlab("") +
  ylab("Percentage of expressed genes") +
  ylim(0,80) +
  prettify +
  scale_x_discrete(labels = c(PLT,peaks,upregu,anti)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_fill_manual(name="", labels = c(PLT,peaks,upregu,anti), values=c("#73D055","#1F968B","#39568C","#440154")) +
  scale_color_manual(name="", labels = c("","","","",""),values=c("black","black","black","black"))
ggsave(paste(output_path,"violin_diff_4.pdf",sep=""), width = 7, height = 7.5)

### Compare only bound+upregulated with non-bound-non-upregulated

# No need to repeat Shapiro-Wilk test, see above. Also skip Bartlett test, since Welch's t-test will be used regardless of variance

## Merge datasets
stat_comp_phl_stem_2 <- rbind(phl_stem_plt, phl_stem_anti)
stat_comp_phl_diff_2 <- rbind(phl_diff_plt, phl_diff_anti)

# Welch's t-tests
t.test(phl_stem_plt$prop, phl_stem_anti$prop, alternative = c("two.sided")) # p-value = 8.911e-07
t.test(phl_diff_plt$prop, phl_diff_anti$prop, alternative = c("two.sided")) # p-value = 0.1052

stat_t_stem_2 <- data.frame(
                dataset = c("PLT"),
                signif = c("p-value = 8.911e-07"),
                yaxis = c(60.5)
)

stat_t_diff_2 <-  data.frame(
                dataset = c("PLT"),
                signif = c("p-value = 0.1052"),
                yaxis = c(16)
)

stat_comp_phl_stem_2$dataset <- factor(stat_comp_phl_stem_2$dataset, levels=c("PLT","anti"))
stat_comp_phl_diff_2$dataset <- factor(stat_comp_phl_diff_2$dataset, levels=c("PLT","anti"))
stat_t_stem_2$dataset <- factor(stat_t_stem_2$dataset, levels=c("PLT","anti"))
stat_t_diff_2$dataset <- factor(stat_t_diff_2$dataset, levels=c("PLT","anti"))

# Plot only bound+upregulated with non-bound-non-upregulated
ggplot(stat_comp_phl_stem_2) +
  geom_violin(aes(x = dataset, y = prop, fill = dataset), color = "black", scale="width") +
  geom_jitter(aes(x = dataset, y = prop, color = dataset), position=position_jitterdodge(jitter.width = 0.2, jitter.height = 0), size = 1) +
  geom_text(data = stat_t_stem_2, aes(x = dataset, y = yaxis, label = signif)) +
  xlab("") +
  ylab("Percentage of expressed genes") +
  ylim(0,80) +
  prettify +
  scale_x_discrete(labels = c(PLT,anti)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_fill_manual(name="", labels = c(PLT,anti), values=c("#73D055","#440154")) +
  scale_color_manual(name="", labels = c("","","","",""),values=c("black","black"))
ggsave(paste(output_path,"violin_stem_2.pdf",sep=""), width = 6, height = 7)

# Plot only bound+upregulated with non-bound-non-upregulated
ggplot(stat_comp_phl_diff_2) +
  geom_violin(aes(x = dataset, y = prop, fill = dataset), color = "black", scale="width") +
  geom_jitter(aes(x = dataset, y = prop, color = dataset),   position=position_jitterdodge(jitter.width = 0.2, jitter.height = 0), size = 1) +
  geom_text(data = stat_t_diff_2, aes(x = dataset, y = yaxis, label = signif)) +
  xlab("") +
  ylab("Percentage of expressed genes") +
  ylim(0,80) +
  prettify +
  scale_x_discrete(labels = c(PLT,anti)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_fill_manual(name="", labels = c(PLT,anti), values=c("#73D055","#440154")) +
  scale_color_manual(name="", labels = c("","","","",""),values=c("black","black"))
ggsave(paste(output_path,"violin_diff_2.pdf",sep=""), width = 6, height = 7)
```

## Build PLT regulome core
```{r}
## Define early and late phloem cells
cell_nr_early <- phloem_sel %>%
  filter(Pseudotime_PSE <= 1 & norm.counts > 5) %>%
  select(Cell_ID) %>%
  unique() %>%
  count() %>%
  as.numeric()

cell_nr_late <- phloem_sel %>%
  filter(Pseudotime_PSE >= 86 & norm.counts > 5) %>%
  select(Cell_ID) %>%
  unique() %>%
  count() %>%
  as.numeric()

## Filter for expression of a gene in at least 10% of cells
phloem_early <- phloem_sel %>%
  filter(Pseudotime_PSE <= 1 & norm.counts > 5) %>%
  select(AGI, Cell_ID) %>%
  mutate(stage = "early") %>%
  select(AGI, stage) %>%
  group_by(AGI, stage) %>%
  summarize(count=n()) %>%
  mutate(prop = count/cell_nr_early) %>%
  filter(prop >= 0.1)
  
phloem_late <- phloem_sel %>%
  filter(Pseudotime_PSE >= 86 & norm.counts > 5) %>%
  select(AGI, Cell_ID) %>%
  mutate(stage = "late") %>%
  select(AGI, stage) %>%
  group_by(AGI, stage) %>%
  summarize(count=n()) %>%
  mutate(prop = count/cell_nr_late) %>%
  filter(prop >= 0.1)

# Compile lists of genes for Venns and heatmap
data.list <- list(SAM = as.matrix(SAMtotvenn$AGI),
                  QC = as.matrix(QCvenn$AGI),
                  Phl_stem = as.matrix(phloem_early$AGI),
                  Mer = as.matrix(Mervenn$AGI),
                  Elo = as.matrix(Elongvenn$AGI),
                  Mat = as.matrix(Matvenn$AGI),
                  Phl_diff = as.matrix(phloem_late$AGI),
                  AC = as.matrix(ACvenn$AGI),
                  BC = as.matrix(BCvenn$AGI))

Phl_stem = as.matrix(phloem_early$AGI)
Mer = as.matrix(Mervenn$AGI)
QC = as.matrix(QCvenn$AGI)
Phl_diff = as.matrix(phloem_late$AGI)
Mat = as.matrix(Matvenn$AGI)
Elo = as.matrix(Elongvenn$AGI)
SAM = as.matrix(SAMtotvenn$AGI)
AC = as.matrix(ACvenn$AGI)
BC = as.matrix(BCvenn$AGI)

ggvenn(data.list, c("Mer","QC","Phl_stem","SAM"), show_percentage = F, text_size = 5, fill_color = c("#bedddc","#fce9c7","#d7e1a6","#e2a6a6"))
ggsave(paste(output_path,"core_venn.pdf",sep=""), width = 5, height = 5)
```

## Check how much of the core is expressed
```{r, fig.width = 3, fig.height = 1.2}
core_stem_PLT_targets <- intersect(intersect(QC, Phl_stem), intersect(SAM, Mer)) %>% as.data.frame()
colnames(core_stem_PLT_targets)[1] <- "AGI"

data.list <- list(core = as.matrix(core_stem_PLT_targets$AGI),
                  Phl_diff = as.matrix(phloem_late$AGI),
                  Mat = as.matrix(Matvenn$AGI),
                  Elo = as.matrix(Elongvenn$AGI),
                  AC = as.matrix(ACvenn$AGI),
                  BC = as.matrix(BCvenn$AGI))

ggvenn(data.list, c("core","AC"), show_percentage = F, text_size = 7, fill_color = c("#A7BED3","#F1FFC4"))
ggvenn(data.list, c("core","BC"), show_percentage = F, text_size = 7, fill_color = c("#A7BED3","#F1FFC4"))
ggvenn(data.list, c("core","Mat"), show_percentage = F, text_size = 7, fill_color = c("#A7BED3","#F1FFC4"))
ggvenn(data.list, c("core","Elo"), show_percentage = F, text_size = 7, fill_color = c("#A7BED3","#F1FFC4"))
ggvenn(data.list, c("core","Phl_diff"), show_percentage = F, text_size = 7, fill_color = c("#A7BED3","#F1FFC4"))

core_overlap <- data.frame(
 tissue = c("AC","BC","Elong","Mat","Phl_diff"),
 overlap = c(146,139,153,114,32)
)
core_overlap$prop <- core_overlap$overlap / 184 * 100

ggplot() +
  geom_bar(data = core_overlap, aes(x = tissue, y = prop), fill = "#ededed", color = "black",stat = "identity") +
  ylim(0,100) +
  xlab("") +
  ylab("Percentage shared with core") +
  prettify +
  scale_x_discrete(labels = rev(c("Phloem diff. cell", "Root mat. zone", "Root elo. zone","Apical cell embryo", "Basal cell embryo"))) +
  coord_flip()
ggsave(paste(output_path,"core_overlap.pdf",sep=""), width = 6, height = 2)

# Which AC/BC genes overlap with the core network?
core_AC_PLT_targets <- intersect(AC, core_stem_PLT_targets$AGI) %>% as.data.frame()
colnames(core_AC_PLT_targets)[1] <- "AGI"
core_BC_PLT_targets <- intersect(BC, core_stem_PLT_targets$AGI) %>% as.data.frame()
colnames(core_BC_PLT_targets)[1] <- "AGI"
```
